src/Files.App/Helpers/Layout/LayoutSizeKindHelper.cs

```diff
@@ -93,6 +93,41 @@ public static int GetListViewRowHeight(ListViewSizeKind listViewSizeKind)
 			}
 		}
 
+		/// <summary>
+		/// Gets the desired icon size for specified layout
+		/// </summary>
+		/// <param name="listViewSizeKind"></param>
+		/// <returns></returns>
+		public static uint GetIconSize(FolderLayoutModes folderLayoutMode)
```

Comment:
This is based on the earlier version in `LayoutPreferencesManager.cs` but has been modified to return the proper icon sizes for the Details and List View.

Reply:
I moved the method out of `LayoutPreferencesManager` as the icon sizes don't get saved to the folders preferences.